salvo/src/lib/test_source_tree.py

```diff
@@ -9,83 +9,240 @@
+    st.get_head_hash()
+
+  assert "No origin is defined or can be" in str(pull_exception.value)
+
 
 def test_git_with_origin():
   """Verify that at a minimum, we can work with a remote origin url
      specified.
   """
-  pass
+  source_repository = proto_source.SourceRepository(
+      source_url='somewhere_in_github'
+  )
+  st = source_tree.SourceTree(source_repository)
+
+  assert st.get_origin()
+
 
 def test_source_tree_with_local_workdir():
   """Verify that we can work with a source location on disk."""
-  pass
+  source_repository = proto_source.SourceRepository(
+      source_path='/some_source_path'
+  )
+  st = source_tree.SourceTree(source_repository)
+
+  remote_string = 'origin  git@github.com:username/reponame.git (fetch)'
+
+  with mock.patch('src.lib.cmd_exec.run_command',
+                  mock.MagicMock(return_value=remote_string)) as magic_mock:
+    origin = st.get_origin()
+    assert origin == "git@github.com:username/reponame.git"
+
+    cmd_params = cmd_exec.CommandParameters(cwd='/some_source_path')
+    magic_mock.assert_called_once_with("git remote -v", cmd_params)
 
 def test_get_origin_ssh():
   """Verify that we can determine the origin for a local repository.
 
   In this instance the repo was cloned via ssh.
   """
-  pass
+  remote_string = 'origin  git@github.com:username/reponame.git (fetch)'
+  gitcmd = "git remote -v"
+
+  source_repository = proto_source.SourceRepository(
+      source_path='/tmp'
+  )
+  st = source_tree.SourceTree(source_repository)
+
+  with mock.patch('src.lib.cmd_exec.run_command',
+                  mock.MagicMock(return_value=remote_string)) as magic_mock:
+    origin_url = st.get_origin()
+
+    cmd_params = cmd_exec.CommandParameters(cwd='/tmp')
+    magic_mock.assert_called_once_with(gitcmd, cmd_params)
+
+    assert origin_url == 'git@github.com:username/reponame.git'
 
 def test_get_origin_https():
   """Verify that we can determine the origin for a local repository.
 
   In this instance the repo was cloned via https
   """
-  pass
+  remote_string = \
+      'origin	https://github.com/aws/aws-app-mesh-examples.git (fetch)'
+  gitcmd = "git remote -v"
+
+  source_repository = proto_source.SourceRepository(
+      source_path='/tmp'
+  )
+  st = source_tree.SourceTree(source_repository)
+
+  with mock.patch('src.lib.cmd_exec.run_command',
+                  mock.MagicMock(return_value=remote_string)) as magic_mock:
+    origin_url = st.get_origin()
+    cmd_params = cmd_exec.CommandParameters(cwd='/tmp')
+    magic_mock.assert_called_once_with(gitcmd, cmd_params)
+
+    assert origin_url == 'https://github.com/aws/aws-app-mesh-examples.git'
 
 def _generate_source_tree_from_origin(origin):
   """Build a source tree from a remote url."""
-  pass
+  source_repository = proto_source.SourceRepository(
+      source_path='/tmp',
+      source_url=origin
+  )
+  return source_tree.SourceTree(source_repository)
 
 def mock_run_command_side_effect(*args):
```

Comment:
Since this is a test helper, future maintainers will need to understand it. Can we unwrap the `*args` into an exact list of arguments and document then and their behavior in the `Args` section of the docstring here?

Applies to all such test helpers below.